### DISCUSSION TITLE
Is there any posibility to register a flow via http?

### DISCUSSION THREAD

**Discussion by rizkyekoputra** [PrefectHQ/prefect#3312] [Q&A] [answered]
If yes, any docs about this?

thanks

**Comment by joshmeek** [comment, marked as answer]
Hi @rizkyekoputra what exactly do you mean by register it over http? Currently registration happens over the GraphQL API by hitting the mutations `create_flow_from_compressed_string` or `create_flow` and those can be sent directly through a POST request.

**Reply by rizkyekoputra** [reply]
yes, thats what I mean @joshmeek. Any docs about how to use it? whats parameter should I fill? cause I am not to familiar using graphql, and there is lack of tutorial for registering flow directly via call the API

**Reply by joshmeek** [reply]
I see, yeah we can definitely expand the documentation around this. There hasn't really been any requests for registering a flow directly over http as most users use the Client's [`.register`](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/client/client.py#L644) function. For some current information on using the GraphQL API check out the docs [here](https://docs.prefect.io/orchestration/concepts/api.html). Here's a part on [registration](https://docs.prefect.io/orchestration/concepts/flows.html#graphql) although it might not be enough information even though I believe only the serialized flow and project ID are required. The `serialized_flow` it accepts is formed by passing in the output from `flow.serialize()`

**Reply by rizkyekoputra** [reply]
oh no, still need calling `flow.serialize()`... I need to run this all on Golang code, currently my backend runs in Golang. You have any suggestion what should I build if I register flow from Golang? Should I create one more service that runs on python than Golang call that python service for all the prefect stuff?

**Reply by joshmeek** [reply]
Yeah you're going to need to get a serialized flow in order to register it with the backend API. Both the `create_flow_from_compressed_string` and `create_flow` require a serialized flow in some form. The serialized flow is simply a JSON representation of the flow metadata so you would probably need some Python service that is able to call `.serialize` on the Python flow objects.

**Reply by rizkyekoputra** [reply]
ok, thanks for the explanations @joshmeek
